pandas/_libs/groupby.pyx

```diff
@@ -1862,12 +1862,16 @@ def group_idxmin_idxmax(
 
     N, K = (<object>values).shape
 
-    group_min_or_max = np.empty_like(out, dtype=values.dtype)
+    if numeric_object_t is object:
+        group_min_or_max = np.empty((<object>out).shape, dtype=object)
```

Comment:
out of curiosity why doesnt the existing line work for object?

Reply:
woops, wrote that comment while looking at the "diff since you last looked".  so the "existing" referred to doing `np.empty_like` unconditionally